server/src/main/java/org/elasticsearch/reservedstate/service/FileSettingsService.java

```diff
@@ -124,7 +126,18 @@ protected void processFileChanges() throws ExecutionException, InterruptedExcept
         ) {
             stateService.process(NAMESPACE, parser, (e) -> completeProcessing(e, completion));
         }
-        completion.get();
+        try {
+            completion.get();
+        } catch (ExecutionException e) {
+            if (e.getCause() instanceof FailedToCommitClusterStateException x) {
+                logger.error("Failed to commit cluster state", x);
+                // This failure is not the fault of FileSettingsService itself.
+                // Throw NoChangeOccurredException to indicate we've already accused the guilty party.
+                throw new NoChangeOccurredException(x);
```

Comment:
I think this will be confusing because by throwing an exception there will still be a log message, but that message must be understood to be "ok". Why do we need to throw an exception at all? It's just caught and logged in `AbstractFileWatchingService.processSettingsAndNotifyListeners()`, which is the caller of this method.

Alternatively, AbstractFileWatchingService could have an abstract method to log errors (taking the exception), and that could then be overriden here and lets this implementation decide how to handle each error.

Reply:
Oh that's an interesting idea. I can try that one out and see how it looks.